C23_WMDE_Mobile_DE_12

moiikana opened the pull request:
https://phabricator.wikimedia.org/T352889

gbirke commented on shared/components/ui/form/SelectCustomAmount.jsx:
```diff
@@ -1,18 +1,68 @@
+			</div>
+		</label>;
+	}
+
```
I'm torn on the implementation. I'd have liked it more if `SelectCustomAmount` had stayed stateless and the blur and value changes and the euro sign would be implemented purely with CSS (see https://spenden.wikimedia.de/ ). The performance would be better and it would be less code. OTOH the JavaScript makes the behavior (or at least the intended styling "hooks" (i.e. CSS classes) very explicit and more readable.

This is not a "please change" recommendation, more like a reminder to use CSS instead of JS if possible.

Abban replied:
Yeah this was a tricky one, some of the required behaviour of the euro symbol depends on js (keeping it visible when the field has a value) so I decided to keep the code that controls that together rather than split it between js/css.

Abban commented on shared/components/ui/form/SelectCustomAmount.jsx:
```diff
@@ -1,18 +1,68 @@
+
+	onFocus = ( e ) => {
+		this.setState( { showEuro: true, focused: true } );
+		this.radioInput.current.click();
```
Yes, I found that it had a weird behaviour where it added the attribute onto the element in the DOM and prevented the user from being able to select the other radio options.

Abban commented on shared/number_formatter/de.js:
```diff
@@ -3,6 +3,7 @@ import amountForServerFormatter from './server';
 
 export const integerFormatter = formatter( { round: 0, integerSeparator: '.' } );
 export const amountInputFormatter = formatter( { round: 2, suffix: ' €', decimal: ',', integerSeparator: '' } );
+export const customAmountInputFormatter = formatter( { round: 2, decimal: ',', integerSeparator: '' } );
```
Is there a preferred style for this? In front of the variable, like a docblock or trailing at the end of the line like in CSS?

gbirke replied:
Like a docblock

moiikana committed:
Archive C23_WMDE_Mobile_DE_11

moiikana committed:
Prepare campaign for C23_WMDE_Mobile_DE_12

- based on mobile de 11 ctrl
- copy changes in both banners, hide link to different yearly amount

https://phabricator.wikimedia.org/T352889

moiikana committed:
Implement VAR for C23_WMDE_Mobile_DE_12

- payment logos
- add tests

https://phabricator.wikimedia.org/T352889

gbirke reviewed (approved):
The extra style file (`C23_WMDE_Mobile_DE_09` should be incorporated as a `Mikings` variant file, e.g. `SoftClose/SoftCloseWithImprovedButton`) to be named more properly and not after the test which gives no indication what it overrides. But you can do that in the mobile-13 PR
